### PR TITLE
Added I2C_1 and CAN_1 master for the Nucleo L476RG board.

### PR DESCRIPTION
**Pull request by Josh0027**
Created Nine Degrees of Freedom Sample Application using MPU6050
	and HMC5883L sensors. Validated on Nucleo L476RG and
	Intel Quark D2000.

Validated the CAN_1 functionality on the Nucleo L476RG.

Signed-off-by: Jose S. Cofreros Jr <josh.cofreros@gmail.com>

### CODE REVIEW

File: drivers/sensor/hmc5883l/Kconfig [hunk 1]
```diff
@@ -20,7 +20,7 @@ config HMC5883L_NAME
 config HMC5883L_I2C_MASTER_DEV_NAME
 	string "I2C master where HMC5883L is connected"
 	depends on HMC5883L
-	default "I2C_0"
+	default "I2C_1"
```
Comment (nashif) [inline drivers/sensor/hmc5883l/Kconfig]: why are you changing this here? If you need that for you sample, you can set to I2C_1 in your sample.

File: boards/arm/nucleo_l476rg/pinmux.c [hunk 2]
```diff
@@ -56,6 +56,12 @@ static const struct pin_config pinconf[] = {
 	{STM32_PIN_PC11, STM32L4X_PINMUX_FUNC_PC11_SPI3_MISO},
 	{STM32_PIN_PC12, STM32L4X_PINMUX_FUNC_PC12_SPI3_MOSI},
 #endif /* CONFIG_SPI_3 */
+
+#ifdef CONFIG_CAN_1
```
Comment (nashif) [inline boards/arm/nucleo_l476rg/pinmux.c]: unrelated change

**Review by nashif — CHANGES_REQUESTED** [review]
please split the changes into multiple commits, can ,sample, dts changes for the boards, etc.

File: boards/arm/nucleo_l476rg/Kconfig.defconfig [hunk 3]
```diff
@@ -35,4 +35,7 @@ config PWM_STM32_2
 
 endif # PWM
 
+config I2C_1
```
Comment (erwango) [inline boards/arm/nucleo_l476rg/Kconfig.defconfig]: This line should be under `if SPI` statement.

File: boards/arm/nucleo_l476rg/nucleo_l476rg_defconfig [hunk 4]
```diff
@@ -39,3 +39,7 @@ CONFIG_PWM=y
 
 #enable MPU
 CONFIG_STM32_ARM_MPU_ENABLE=y
+
+#enable i2c
+CONFIG_I2C_1=y
```
Comment (erwango) [inline boards/arm/nucleo_l476rg/nucleo_l476rg_defconfig]: This is not needed if alredy done in Kconfig.defconfig

File: samples/sensor/nine_degrees_freedom/src/main.c [hunk 5]
```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2012-2014 Wind River Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * 08/23/2018	Jose Sune Cofreros Jr. 	Created nine degrees of freedom
```
Comment (erwango) [inline samples/sensor/nine_degrees_freedom/src/main.c]: Please reuse SPDX boilerplate

File: samples/sensor/nine_degrees_freedom/src/main.c [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+#include <zephyr.h>
+#include <device.h>
+#include <sensor.h>
+#include <stdio.h>
+
+void main(void)
+{
+	struct device *dev1 = device_get_binding("MPU6050");
+	struct device *dev2 = device_get_binding("HMC5883L");
+
```
Comment (erwango) [inline samples/sensor/nine_degrees_freedom/src/main.c]: you should check dev1 and dev2 are not NULL

File: boards/arm/nucleo_l476rg/Kconfig.board [hunk 7]
```diff
@@ -9,3 +9,5 @@
 config BOARD_NUCLEO_L476RG
 	bool "Nucleo L476RG Development Board"
 	depends on SOC_STM32L476XG
+	select HAS_DTS
```
Comment (erwango) [inline boards/arm/nucleo_l476rg/Kconfig.board]: Not needed. Done in arch/arm/code/Kconfig

File: boards/arm/nucleo_l476rg/Kconfig.board [hunk 8]
```diff
@@ -9,3 +9,5 @@
...
+	select HAS_DTS_I2C
```
Comment (erwango) [inline boards/arm/nucleo_l476rg/Kconfig.board]: Not needed. Done in `drivers/i2c/Kconfig.stm32`

**Review by erwango — CHANGES_REQUESTED** [review]
Some additional comments

File: boards/arm/nucleo_l476rg/Kconfig.defconfig (hunk 3 above)
Comment (Josh0027) [reply boards/arm/nucleo_l476rg/Kconfig.defconfig]: Added under "if I2C" instead.

File: boards/arm/nucleo_l476rg/nucleo_l476rg_defconfig (hunk 4 above)
Comment (Josh0027) [reply boards/arm/nucleo_l476rg/nucleo_l476rg_defconfig]: Reverted to previous code.

File: boards/arm/nucleo_l476rg/pinmux.c (hunk 2 above)
Comment (Josh0027) [reply boards/arm/nucleo_l476rg/pinmux.c]: Removed all instances of CAN.

File: drivers/sensor/hmc5883l/Kconfig (hunk 1 above)
Comment (Josh0027) [reply drivers/sensor/hmc5883l/Kconfig]: Reverted to original code and declared in prj.conf.

File: samples/sensor/nine_degrees_freedom/src/main.c (hunk 5 above)
Comment (Josh0027) [reply samples/sensor/nine_degrees_freedom/src/main.c]: Done. Followed common boilerplate.

File: samples/sensor/nine_degrees_freedom/src/main.c (hunk 6 above)
Comment (Josh0027) [reply samples/sensor/nine_degrees_freedom/src/main.c]: Done.

**Comment by Josh0027** [comment]
Hello. Sorry to bother but can anyone reject my last pull request? I suddenly have all of these non-related changes on the code review. I think I made a mistake on not deleting all the other commits while I was following the review request instructions.

**Comment by Josh0027** [comment]
pfalcon, thank you. I will just close this request and create a new one.

**Comment by jukkar** [comment]
Instead of closing, you could just have force pushed a rebased branch.

**Comment by pfalcon** [comment]
> Instead of closing, you could just have force pushed a rebased branch.

I'm not sure if that will clear folks superfluously and automatically added as reviewers, so that would be a reason to open a new one. But otherwise, yes, force-pushing a new version is always preferred, because it preserves all the discussion already held. @Josh0027, if you decide to open a new one in the end, please be sure to link back to this one. Thanks!

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Comment by pfalcon** [comment]
@Josh0027: So I guess, it's better to close this one after all, because it confuses people who got into the reviewer list. Please submit a new, cleaned one. Thanks.

@dbkinder : Sorry for the hassle, this is the case of "stray commits got into PR by mistake".
